system/jlib/jstats.cpp

```diff
@@ -2091,7 +2091,35 @@ class TreeNodeStatisticGatherer : public CInterfaceOf<IStatisticGatherer>
 
 void CRuntimeStatistic::merge(unsigned __int64 otherValue, StatsMergeAction mergeAction)
 {
-    value = mergeStatisticValue(value, otherValue, mergeAction);
+    switch (mergeAction)
+    {
+    case StatsMergeKeepNonZero:
+        if (otherValue)
```

Comment:
@shamser I wonder if this should say

if (otherValue && !value)

to avoid the more expensive cas if the value has already been set.
I am happy to merge as it is.